src/main/java/com/ProjectEureka/backend/impls/UserServiceImpls.java

```diff
@@ -0,0 +1,72 @@
+package com.ProjectEureka.backend.impls;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.ProjectEureka.backend.models.User;
+import com.ProjectEureka.backend.repositories.UserRepository;
+import com.ProjectEureka.backend.services.UserService;
+import org.springframework.stereotype.Service;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+
+@Service
+public class UserServiceImpls implements UserService {
+    @Autowired
+    UserRepository userRepository;
+
+    @Override
+    public User createUser(User user) {
+        return userRepository.save(user);
+    }
+
+    @Override
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public User getUserById(String id) {
+        Optional<User> userDb = this.userRepository.findById(id);
+
+        if (userDb.isPresent()) {
+            return userDb.get();
+        } else {
+            throw new ResourceNotFoundException("User not found with id: " + id);
+        }
+    }
+
+    @Override
+    public void deleteById(String id) {
+        Optional<User> deletedUser = this.userRepository.findById(id);
+
+        if (deletedUser.isPresent()) {
+            this.userRepository.delete(deletedUser.get());
```

Comment:
consider returned the deleted user, it'd help the front-end view which user get's deleted. 

This would also be a good appliance of returning a User named`deletedUser`